### ISSUE TITLE
Spread isn't properly checked

### ISSUE THREAD

**Issue by deser** [microsoft/TypeScript#43964]
# Bug Report

### 🔎 Search Terms
 optional, spread

### 🕗 Version & Regression Information

4.2.3

### ⏯ Playground Link
https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBASwHY2FAZgQwMbDgQTDABsBPfJUgdwAs1gARTGTOAbwCg5u4ArCAEZMWAfgBc7AL4BuLj0gRiwzOKmyecTJVr0AyixgBXAM6q2Mud0xEyAFVJhgZixuOZimKKQCi4YNgMECCRTCXN1eQhjYwQBYmB9TxhhJwljGChkAHMIqxtSAAUoYHR6JFxQuHTMpBzLOCpMBBhsgrQggBNVJEMAWwE0XMRjCmDSXogTVQEIRWAtIcMYXrN6jQBtAGtgUkQkKozsgF1Vauyhlx5DKGJTw9qhjuZgAFUwJ9QuiR7+wfqP4AAYWKzy+cB+AygQ2KvH8LWCKW6fUhskkHA4oEgsD2qAwODwhBIu04GgQHTS9zqGhMaAAkuSDjUqTx0sAwBSmXAAD5wQxIDolZDADpDGgIJkAGWQmw52W5vP5gqQwqGyFxADcEMAqMowpJ5XyBeghSL6lpqHRirqCPkKBb6MoDYrjcrTRoAcD5p8kb8oRw0RxsMF0pp8hJCWQ4ABedhyMkSADkmGMHQTABp6jSoPTE8nUxnXKh2XAkyn0-UxZLpbmywWeGq0JrtdbDUrhXWrNpLYxmJgJAqjSaO3APSDPhIAEwAZgALNOZ-70RwAPTLuC+RwBYWafZoKDQODdwPB+DYLS4eIdCPkLsO3vh223q296Ox7gAOk-1iJGZkcCXq7cAAxAAjBIe4Ht2wxwEGUDFAEgbnsAl7XnaOjPiw75ZvSr4Jm4HToAmsgrmucBARO4FwZB9DQbB8EwMeISnkhKGPvaGGYGBNpEmh3aOjGJK8sYdIMgmvTABAVAJv60hwEAA

### 🙁 Actual behavior
TS doesn't throw error on incorrect spread (look into PG link)

### 🙂 Expected behavior
TS throws error on incorrect spread (look into PG link)

**Comment by RyanCavanaugh** [comment]
This is the intended behavior. Spreads are allowed to contribute excess properties.

**Comment by deser** [comment]
Look: the same action in JS done in different ways treated by TS differently:
1.  userId will appear here and IT IS OK
![image](https://user-images.githubusercontent.com/3010698/117173572-92f37480-add5-11eb-8dbb-87f1a827d611.png)

2. When I try to set userId explicitly TS errors:
![image](https://user-images.githubusercontent.com/3010698/117173653-a69edb00-add5-11eb-8b2e-d594494905fe.png)

Result in JS eventually is the same: userId will appear in canceledApplyAnywhereData which SHOULD NOT happen as such property is not declared in ApplyAnywhereData.

**Comment by tjjfvi** [comment]
Object types in TS are unstrict:
```ts
declare const a: { x: number, y: boolean }
const b: { x: number } = a // Ok
```
Object *literal* types are checked strictly:
```ts
const b: { x: number } = {
  x: 5,
  y: true, // Error, we probably didn't mean to do this
}
  ```

**Comment by chriskuech** [comment]
> This is dangerous.

This has and will continue to cause the very kinds of bugs that the typechecker should be preventing. I hope this can be reconsidered in the future considering how nice of a feature spread is yet has been effectively banned on my team for this behavior.
